evm/src/fuzz/invariant/executor.rs

```diff
@@ -407,6 +407,25 @@ impl<'a> InvariantExecutor<'a> {
             ["targetSenders", "excludeSenders", "targetContracts", "excludeContracts"]
                 .map(|method| self.get_list::<Address>(invariant_address, abi, method));
 
+        let proxies = self.get_list::<(Address, String)>(invariant_address, abi, "targetProxies");
+
+        let mut combined: TargetedContracts = BTreeMap::new();
+
+        for (key, identifier) in &proxies {
+            // Borrow proxies to avoid taking ownership
+            if let Some((_, (abi, _))) =
+                self.project_contracts.find_by_name_or_identifier(identifier)?
+            {
+                // Merge the functions of multiple contracts with the same address
+                combined
+                    .entry(*key)
+                    .or_insert_with(|| (identifier.clone(), abi.clone(), vec![]))
+                    .1
+                    .functions
+                    .extend(abi.functions.clone());
+            }
+        }
```

Comment:
this looks right, would love to see some additional docs on what exactly is happening here

Reply:
Think it a lot better now. Reworked to destructure the entry and avoid `.1.` and expanded the documentation.